scripts/o3de/tests/test_export_project.py

```diff
@@ -0,0 +1,152 @@
+    pytest.param(TEST_PYTHON_SCRIPT, [456], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 0),
+    # failure cases
+    pytest.param(TEST_PYTHON_SCRIPT, [], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 1),
+    pytest.param(TEST_PYTHON_SCRIPT, [456], False, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), pathlib.PurePath("test_output.txt"), False, check_for_o3de_context_arg, 1),
+    # TEST_ERR_PYTHON_SCRIPT
+    pytest.param(TEST_ERR_PYTHON_SCRIPT, [], True, pathlib.PurePath("test_project"), pathlib.PurePath("export_scripts"), None, True, None, 1)
+])
+def test_export_script(tmp_path,
+                       input_script, 
+                       args, 
+                       should_pass_project_folder, 
+                       project_folder_subpath, 
+                       script_folder_subpath, 
+                       output_filename, 
+                       is_expecting_error, 
+                       output_evaluation_func,
+                       expected_result):
+    import sys
+
+    project_folder = tmp_path / project_folder_subpath
+    project_folder.mkdir()
+
+    script_folder = tmp_path / script_folder_subpath
+    script_folder.mkdir()
+
+
+    project_json = project_folder / "project.json"
+    project_json.write_text(TEST_PROJECT_JSON_PAYLOAD)
+
+
+    test_script = script_folder / "test.py"
+    test_script.write_text(input_script)
+
+    if output_filename:
+        test_output = script_folder / output_filename
+
+        assert not test_output.is_file()
+    
+    result = _export_script(test_script, project_folder if should_pass_project_folder else None, args)
+
+    assert result == expected_result
+
+
+    # only check for these if we're simulating a successful case
+    if result == 0 and not is_expecting_error:
+        assert str(script_folder) in sys.path
+
+        if output_filename:
+            assert test_output.is_file()
+
+            with open(test_output, 'r') as t_out:
+                output_text = t_out.read()
+        
+            if output_evaluation_func:
+                output_evaluation_func(output_text, args)
+
+        o3de_cli_folder = pathlib.Path(__file__).parent.parent / "o3de"
+        found_path = False
+
+        # had to change this to use pathlib instead b/c windows path's have a subtle difference for the C drive
+        # the pathlib library capitalizes it as 'C' in the string repr, while python uses lowercase 'c'
+        for sp in sys.path:
+            found_path = found_path or o3de_cli_folder == pathlib.Path(sp)
+            if found_path: break 
+        assert found_path
```

Comment:
This could be simplified to
```suggestion
        assert o3de_cli_folder in [pathlib.Path(sysPath) for sysPath in sys.path]
```